lib/resource-providers/utils.ts

```diff
@@ -0,0 +1,27 @@
+import { IResource } from 'aws-cdk-lib/core';
+import { DummyProxy } from '../utils/proxy-utils';
+import { ResourceContext } from '../spi';
+import { v4 as uuidv4 } from 'uuid';
+
+export function getNamedResource<T extends IResource = IResource>(resourceName : string) : T {
```

Comment:
Could we add couple of lines of documentation of what this function is doing?

Reply:
addressed